tests/vec.rs

```diff
@@ -90,9 +82,9 @@ fn insert() {
     particles.push(Particle::new(String::from("Na"), 0.0));
 
     particles.insert(1, Particle::new(String::from("Zn"), 0.0));
-    assert_eq!(particles.name[0], "Cl");
```

Comment:
I've actually looked a bit into it and was not straightforward. But I agree that this would make for a much better interface. Even better, the code would be the same as using `Vec`.

Reply:
Actually after looking into `SliceIndex` it actually seems doable.